dropwizard-logging/src/test/java/io/dropwizard/logging/LoggingFactoryPrintErrorMessagesTest.java

```diff
@@ -0,0 +1,93 @@
+package io.dropwizard.logging;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.core.util.StatusPrinter;
+import com.codahale.metrics.MetricRegistry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LoggingFactoryPrintErrorMessagesTest {
+
+    @Rule
+    public final TemporaryFolder tempDir = new TemporaryFolder();
+
+    File folderWithoutWritePermission;
+    File folderWithWritePermission;
+
+    LoggingFactory factory;
+    LoggerContext loggerContext;
+
+    @Before
+    public void setUp() throws Exception {
+        folderWithoutWritePermission = tempDir.newFolder("folder-without-write-permission");
```

Comment:
We don't need both folders in one test. I think it's more appropriate just to call `tempDir.newFolder` in every test 
and don't expose `folderWithoutWritePermission` and `folderWithWritePermission` as fields.